core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -326,6 +326,14 @@ private void saveSssdIpaPillar(Cluster cluster, Map<String, SaltPillarProperties
             sssdConnfig.put("domain", kerberosConfig.getDomain());
             sssdConnfig.put("password", kerberosConfig.getPassword());
             sssdConnfig.put("server", kerberosConfig.getUrl());
+            Map<String, List<String>> serviceLocations = getServiceLocations(cluster, blueprintService.isAmbariBlueprint(cluster.getBlueprint()));
+            // enumeration has performance impacts so it's only enabled if Ranger is installed on the cluster
+            // otherwise the usersync does not work with nss
+            if (CollectionUtils.isEmpty(serviceLocations.get("RANGER_ADMIN"))) {
```

Comment:
Or simply `sssdConnfig.put("enumerate", !CollectionUtils.isEmpty(serviceLocations.get("RANGER_ADMIN")));`. 😃

Reply:
Right, I'll make the change

Reply:
fixed

Reply:
Thanks. 😃

Reply:
Thanks for the review!